lib/workers/repository/update/branch/get-updated.ts

```diff
@@ -361,6 +366,26 @@ export async function getUpdatedPackageFiles(
   };
 }
 
+// workaround, see #27319
+function patchConfigForArtifactsUpdate(
+  config: BranchConfig,
+  manager: string,
+  packageFileName: string,
+): BranchConfig {
+  const updatedConfig = { ...config };
+  if (is.nonEmptyArray(updatedConfig.packageFiles?.[manager])) {
+    const managerPackageFiles: PackageFile[] =
+      updatedConfig.packageFiles?.[manager];
+    const packageFile = managerPackageFiles.find(
+      (p) => p.packageFile === packageFileName,
+    );
+    if (packageFile) {
+      updatedConfig.lockFiles = packageFile.lockFiles;
```

Comment:
```suggestion
      updatedConfig.lockFiles ??= packageFile.lockFiles;
```

This is just a little more defensive/less likely to break things in future